File: csharp/CsFind/CsFindTests/FileResultTests.cs
using NUnit.Framework;
using CsFindLib;

namespace CsFindTests;

[TestFixture]
class FileResultTests
{
	private const string CsFinderPath = "~/src/xfind/csharp/CsFind/CsFind/Finder.cs";
	// This is temporary until all versions support preserving tilde in path
	private readonly string _expandedCsFinderPath = $"{FileUtil.GetHomePath()}/src/xfind/csharp/CsFind/CsFind/Finder.cs";

	[Test]
	public void FileResult_ToString_EqualsExpected()
	{
		var fileResult = new FileResult(CsFinderPath, FileType.Text);
		// Assert.That(fileResult.ToString(), Is.EqualTo(CsFinderPath));
		Assert.That(fileResult.ToString(), Is.EqualTo(_expandedCsFinderPath));
	}

	[Test]
	public void FileResultTrailingSlash_ToString_EqualsExpected()
	{
		var fileResult = new FileResult(CsFinderPath, FileType.Code);
		// Assert.That(fileResult.ToString(), Is.EqualTo(CsFinderPath));
		Assert.That(fileResult.ToString(), Is.EqualTo(_expandedCsFinderPath));
	}
}

File: fsharp/FsFind/FsFindTests/FileResultTests.fs
namespace FsFindTests

open System.IO
open NUnit.Framework
open FsFindLib

[<TestFixture>]
type FileResultTests () =

    member this.CsFindPath = "~/src/xfind/csharp/CsFind/CsFind"
    member this.WinCsFindPath = @"C:\src\xfind\csharp\CsFind\CsFind"

    [<SetUp>]
    member this.Setup () =
        ()

    [<Test>]
    member this.FileResult_ToString_EqualsExpected () =
        let fi = Path.Join(this.CsFindPath, "Finder.cs")
        let fileType = FileType.Code
        let fr = FileResult.Create fi fileType 0L None
        Assert.That(FileResult.ToString(fr), Is.EqualTo($"%s{this.CsFindPath}/Finder.cs"))
        ()

    [<Test>]
    member this.FileResultTrailingSlash_ToString_EqualsExpected () =
        let fi = Path.Join($"%s{this.CsFindPath}/", "Finder.cs")
        let fileType = FileType.Code
        let fr = FileResult.Create fi fileType 0L None
        Assert.That(FileResult.ToString(fr), Is.EqualTo($"%s{this.CsFindPath}/Finder.cs"))
        ()
